ndcube/mixins/plotting.py

```diff
@@ -329,16 +333,20 @@ def _plot_3D_cube(self, plot_axis_indices=None, axes_coordinates=None,
                 data = data.reshape(new_shape)
             # Generate plot
             ax = ImageAnimatorWCS(data, wcs=self.wcs, image_axes=plot_axis_indices,
-                                  unit_x_axis=axes_units[plot_axis_indices[0]],
-                                  unit_y_axis=axes_units[plot_axis_indices[1]],
-                                  axis_ranges=axes_coordinates, **kwargs)
+                                  unit_x_axis=new_axes_units[plot_axis_indices[0]],
+                                  unit_y_axis=new_axes_units[plot_axis_indices[1]],
+                                  axis_ranges=new_axes_coordinates, **kwargs)
+
+            # ax.axes.coords[0].set_axislabel(self.wcs.world_axis_physical_types[plot_axis_indices[0]])
```

Comment:
This is the code commented for the labeling, some of the examples are failing, so that needs to be seen.

Reply:
Done, the label part is working!